pkg/operator/gcwatchercontroller/prometheus_client.go

```diff
@@ -0,0 +1,75 @@
+package gcwatchercontroller
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/transport"
+
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/operatorclient"
+)
+
+func newPrometheusClient(ctx context.Context, secretClient corev1client.SecretsGetter, configMapClient corev1client.ConfigMapsGetter) (prometheusv1.API, error) {
```

Comment:
I think that this isn't the right approach, it's probably ok for tests, but not for a production use. Would be nice to double check with @s-urbaniak about the right way to get the correct token.

Reply:
I am using our serviceaccount token now, same as [etcd operator](https://github.com/openshift/cluster-etcd-operator/blob/c9977ae3bd788a9e9a595001c5bc24995f9d0175/pkg/operator/metriccontroller/client.go) does

Reply:
Yeah, this approach is definitely better.